agent_plugin_builder/build_plugin.py

```diff
@@ -0,0 +1,252 @@
+import json
+import logging
+import tarfile
+from importlib import import_module
+from pathlib import Path
+
+import yaml
+from monkeytypes import AgentPluginManifest
+
+from .build_options import PlatformDependencyPackagingMethod, parse_agent_plugin_build_options
+from .vendor_dirs import (
+    check_if_common_vendor_dir_possible,
+    generate_common_vendor_dir,
+    generate_requirements_file,
+    generate_vendor_dirs,
+)
+
+logger = logging.getLogger(__name__)
+
+
+BUILD = "build"
+DIST = "dist"
+CONFIG_SCHEMA = "config-schema.json"
+SOURCE = "source"
+MANIFEST = "manifest"
+
+
+def build_agent_plugin(
+    plugin_path: Path,
+    build_dir_path: Path,
+    dist_dir_path: Path,
+):
+    """
+    Build the agent plugin by copying the plugin code to the build directory,
+    generating the Agent Plugin archive.
+
+    :param plugin_path: Path to the plugin code.
+        If the directory does not exist, an error will be raised
+    :param build_dir_path: Path to the build directory.
+        If the directory does not exist, it will be created else it will be cleared
+    :param dist_dir_path: Path to the dist directory.
+        If the directory does not exist, it will be created
+    """
+    import shutil
+
+    if not plugin_path.exists():
+        logger.error(f"Plugin path {plugin_path} does not exist")
+        raise FileNotFoundError(f"Plugin path {plugin_path} does not exist")
+
+    if not build_dir_path.exists():
+        logger.info(f"Creating build directory: {build_dir_path}")
+        build_dir_path.mkdir(exist_ok=True)
+    else:
+        try:
+            logger.info(f"Clearing build directory: {build_dir_path}")
+            shutil.rmtree(build_dir_path)
+        except shutil.Error as err:
+            logger.error(f"Unable to clear build directory: {build_dir_path}")
+            raise err
+
+    try:
+        logger.info(f"Copying plugin code to build directory: {plugin_path} -> {build_dir_path}")
+        shutil.copytree(plugin_path, build_dir_path, dirs_exist_ok=True)
+    except shutil.Error as err:
+        logger.error(
+            f"Unable to copy plugin code to build directory: {plugin_path} -> {build_dir_path}"
+        )
+        raise err
+
+    agent_plugin_manifest = get_agent_plugin_manifest(build_dir_path)
+    create_agent_plugin_archive(build_dir_path, agent_plugin_manifest, dist_dir_path)
+
+
+def get_agent_plugin_manifest(build_dir_path: Path) -> AgentPluginManifest:
+    """
+    Get the Agent Plugin manifest from the build directory.
+
+    :param build_dir_path: Path to the build directory.
+    """
```

Comment:
This could raise some kind of yaml error, and/or a file not found error (or similar).